synapse/handlers/room_member.py

```diff
@@ -218,6 +220,52 @@ def _local_membership_update(
                 newly_joined = prev_member_event.membership != Membership.JOIN
             if newly_joined:
                 yield self._user_joined_room(target, room_id)
+
+            # Copy over direct message status and room tags if this is a join
+            # on an upgraded room
+
+            # Check if this is an upgraded room
+            state_ids = yield self.store.get_current_state_ids(room_id)
+            create_id = state_ids.get((EventTypes.Create, ""))
+            if not create_id:
+                return
+            create_event = yield self.store.get_event(create_id)
+
+            if "predecessor" in create_event["content"]:
```

Comment:
presumably we can use the method from the search PR for this

Reply:
Ah, we can but it's inside `SearchHandler`: https://github.com/matrix-org/synapse/blob/b1b6dba2d2c0690157cdd620063d4e16a583264a/synapse/handlers/search.py#L41-L56

Shall I move it to the store instead?

Reply:
I was thinking specifically of `get_room_predecessor`, which is in the Store I think?

Reply:
Oh, so it is. Will add, thanks.